ArduCopter/system.cpp

```diff
@@ -256,12 +256,20 @@ void Copter::update_dynamic_notch()
 
 #if RPM_ENABLED == ENABLED
         case HarmonicNotchDynamicMode::UpdateRPM: // rpm sensor based tracking
-            float rpm;
-            if (rpm_sensor.get_rpm(0, rpm)) {
-                // set the harmonic notch filter frequency from the main rotor rpm
-                ins.update_harmonic_notch_freq_hz(MAX(ref_freq, rpm * ref / 60.0f));
+            if (ins.has_harmonic_option(HarmonicNotchFilterParams::Options::DynamicHarmonic)) {
```

Comment:
It's a shame about the code duplication but I guess its going to be difficult to refactor. You will need to add similar code to plane I believe.

Reply:
Looking at how plane does it, it seems like in some cases the notches track quadplane throttle, while in others it tracks main airplane RPM. Should I be adding a case for either option? Also I don't have a quadplane or plane to test with unfortunately.

Reply:
Okay I've added the plane based version. I agree that the code duplication is annoying. Since ESC telem updates the RPM_ESC_Telem backend, would it be possible to remove the ESC telem case in the switch statement and run the RPM sensor code at a higher rate consistent with the ESC telem backend?

Reply:
Looking at the ESC_Telem backend for RPM harder it seems like it just averages all of the ESC's RPMs. Since ESC telem has a max number of ESCs set to 12 and the RPM lib is currently limited to 2 all the parameters for RPM would have to be extended out to 12 and 12 backends would have to be created. I'm sure there's a better way to do it that could consolidate esc telem and rpm reading into one case in the harmonic notch switch statement, but I'm not exactly sure what the best way is.